### PR TITLE
[BugFix] Fix assigning wrong type for empty array/map/struct

### LINKED ISSUE #16659: [Bug] Be crash on type array<null>
### Steps to reproduce the behavior (Required)

```sql
CREATE TABLE testtesttest
(
    id BIGINT(20) NOT NULL,
    f1 VARCHAR(200) NOT NULL,
    f2 VARCHAR(200) NOT NULL,
    json1 JSON NULL,
    json2 JSON NULL,
    arr1 ARRAY<VARCHAR(200)> NULL,
    arr2 ARRAY<VARCHAR(200)> NULL
) ENGINE = OLAP PRIMARY KEY(id)
COMMENT "test"
DISTRIBUTED BY HASH(id) BUCKETS 10
PROPERTIES (
"replication_num" = "1",
"bloom_filter_columns" = "id, f1, f2",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);

INSERT INTO testtesttest(id, f1, f2, json1, json2, arr1, arr2)
SELECT 2, 'BBB', 'bbb', NULL, NULL, [], [];
```

### Expected behavior (Required)

success

### Real behavior (Required)

be crash

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

+-------------------+
| current_version() |
+-------------------+
| UNKNOWN 350b5a051 |
+-------------------+

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16659

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Type.java [hunk 1]
```diff
@@ -133,6 +133,7 @@ public abstract class Type implements Cloneable {
     public static final PseudoType ANY_MAP = PseudoType.ANY_MAP;
     public static final PseudoType ANY_STRUCT = PseudoType.ANY_STRUCT;
 
+    public static final Type ARRAY_NULL = new ArrayType(Type.NULL);
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:136]: Emm, only used in the hack logic

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Type.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Type.java]: Yes, you are right

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|2|50.00%|[315]|

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Review by Seaven — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 13 (92.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PolymorphicFunctionAnalyzer.java|4|5|80.00%|[113]|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**